web_scraping/steam_scraper/scrape_steam.py

```diff
@@ -67,20 +68,85 @@ def parse_price(game_listing: bs4.Tag) -> str:
     return format_price(price)
 
 
+def get_steam_app_url(app_id: str) -> str:
+    """Formats the base steam url for a particular app."""
+    return f"{STEAM_APP_URL}{app_id}/"
+
+
+def scrape_game_description(app_soup: bs4.BeautifulSoup) -> str:
+    """Scrapes a game's Steam app page for description information."""
+    description_tag = app_soup.find('div', class_='game_description_snippet')
+    if description_tag:
+        return description_tag.text.strip()
+    return "No description available."
+
+
+def scrape_game_tags(app_soup: bs4.BeautifulSoup) -> list[str]:
+    """Scrapes a game's Steam app page for tags."""
+    tag_elements = app_soup.find_all('a', class_='app_tag')
+    tags = [tag.text.strip() for tag in tag_elements]
+    return tags if tags else ["No tags available."]
+
+
+def scrape_game_genres(app_soup: bs4.BeautifulSoup) -> list[str]:
+    """Scrapes a game's Steam app page for genres."""
+    genres_section = app_soup.find('div', id='genresAndManufacturer')
+    if genres_section:
+        genre_elements = genres_section.find_all('a', href=True)
+        genres = [genre.text.strip()
+                  for genre in genre_elements if "genre" in genre['href']]
+    return genres if genres else []
+
+
+def scrape_game_operating_systems(app_soup: bs4.BeautifulSoup) -> list[str]:
+    """Scrapes a game's Steam app page for supported operating systems."""
+    platform_mapping = {
+        'win': 'Windows',
+        'mac': 'Mac',
+        'linux': 'Linux'
+    }
+    platforms = []
+    platform_elements = app_soup.find_all('span', class_='platform_img')
+
+    for platform in platform_elements:
+        if 'class' in platform.attrs and len(platform['class']) > 1:
+            platform_class = platform['class'][1]
+            if platform_class in platform_mapping:
+                platforms.append(platform_mapping[platform_class])
+
+    return platforms if platforms else []
+
+
 def parse_game_listing(game_listing: bs4.Tag) -> dict:
     """Parses game information from a game listing div."""
+    app_id = parse_app_id(game_listing)
+    steam_app = get_steam_app_url(app_id)
+    steam_app_page_source = load_page_source(steam_app)
+    app_soup = bs4.BeautifulSoup(steam_app_page_source, 'html.parser')
+
+    description = scrape_game_description(app_soup)
+    os = scrape_game_operating_systems(app_soup)
+    genres = scrape_game_genres(app_soup)
+    tags = scrape_game_tags(app_soup)
+
     return {
-        'app_id': parse_app_id(game_listing),
-        'url': parse_game_url(game_listing),
-        'img_url': parse_image_url(game_listing),
         'title': parse_title(game_listing),
+        'description': description,
```

Comment:
Could this be shortened by  'description': scrape_game_description(app_soup) . Not overly important, just a suggestion for aesthetics.